platform-sdk/docs/proposals/TSS-Library/TSS-Library.md

```diff
@@ -0,0 +1,808 @@
+When a message is created, all encrypted values are included. Only the intended receivers can decrypt their respective portions of the secret element.
+This setup allows participants to share secret information securely.
+The message also contains additional information necessary for signature validation.
+
+Upon receiving a threshold number of messages, each participant decrypts the information encrypted with their public key, aggregates it, and generates a private key for each owned share.
+They also retrieve a public key for each share in the system to validate signatures.
+Individual signing can now begin. Participants use the private information of their shares to sign messages.
+
+When signatures from at least threshold number of parties are combined, an aggregate signature is created. This aggregate signature can be validated using the combined value of the public shares in the directory.
+The process restarts whenever the number of participants or the shares assigned to each change.
+However, the initially generated group public key remains unchanged to maintain consistency. New secret information for shares is created using existing data, ensuring that the aggregate signature can still be verified with the original group public key.
+
+#### Implementation details
+Before starting, all participants should agree on a `SignatureSchema` they will use.
+The `SignatureSchema` defines the type of Curve and which Group of the Pairing is used for PublicKey Generation.
+
+##### Input
+* Participant's persistent EC Private key (Private to each participant)
+* Number of participants (Public)
+* Number of shares per participant (Public)
+* A threshold value
+* All participants' persistent EC public keys (Public)
+* A predefined `SignatureSchema` (Public / Constant for all the network)
+
+```
+`Share`: An abstract concept having a unique identifier and an owner
+|-  `PrivateShare`: Represents a share owned by the executor of the scheme. Contains a secret value (EC Private key) used for signing.
+|-  `PublicShare`: Represents a share in the system. It contains public information that can be used to validate each signatures.   
+```
+
+#####  Bootstrap Stage
+Given
+
+A participants directory:
+```
+e.g:
+P   # shares
+-----------------------------
+P₁  5        P₁_EC_PublicKey
+P₂  2        P₂_EC_PublicKey
+P₃  1        P₃_EC_PublicKey
+P₄  2        P₄_EC_PublicKey
+
+```
+
+A threshold value:
+`e.g: t = 5`
+
+First, a `shareId`: `sid` is generated for each share. It's a unique, contiguous starting from 1 identifier for each existent share. It is deterministic because the function we use to generate it is deterministic.
+It is necessary to assure they are values that: a) are unique per share, b) non-0, and c) can be used as input for the polynomial (They are from the same field of the selected curve)
```

Comment:
The share IDs are part of the proof and commitment. All nodes have to use the same values for the shared IDs. A malicious node using different values will be detected when validating the tssMessage against the proof and the commitment.
But if there exists a specific set of values such as they would affect the validation, making it take longer than it should, I will defer the answer to @rsinha.

Reply:
As for security, adversarial nodes are free to choose low or zero entropy values for the protocol. There will still be sufficient entropy in the entire protocol, from the honest nodes. As for correctness, the choice of the value does not impact whether any downstream checks fail, as long as the protocol is followed -- if a node deviates from the protocol and produces invalid messages, those are certainly detected. Happy to expand further.

Reply:
Thank you for the explanation, that clarifies somewhat.

It is not certain (there may be "show-the-way" concerns), but it might be helpful to include a section on how this proposed approach handles malicious actors; and include answers like this in that section.
Just a thought.

Reply:
Done